### ISSUE TITLE
[CI] Many tests fail in release building, because missing TSDB index setting.

### ISSUE THREAD

**Issue by martijnvg** [elastic/elasticsearch#80822]
Many tests fail during release build because `index.time_series.end_time` index setting isn't registered.

This index setting probably needs to be behind the tsdb feature flag.

**Build scan:**
https://gradle-enterprise.elastic.co/s/ol6kvzhwrw5la/tests/:modules:analysis-common:test/org.elasticsearch.analysis.common.CJKFilterFactoryTests/testNoFlags

**Reproduction line:**
`./gradlew ':modules:analysis-common:test' --tests "org.elasticsearch.analysis.common.CJKFilterFactoryTests.testNoFlags" -Dtests.seed=957DF4F46E100008 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=mk -Dtests.timezone=Asia/Amman -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.analysis.common.CJKFilterFactoryTests&tests.test=testNoFlags

**Failure excerpt:**
```
java.lang.IllegalArgumentException: Setting is not registered for key [index.time_series.end_time]

  at __randomizedtesting.SeedInfo.seed([957DF4F46E100008:361C91423FE78B96]:0)
  at org.elasticsearch.common.settings.AbstractScopedSettings.addSettingsUpdateConsumer(AbstractScopedSettings.java:208)
  at org.elasticsearch.common.settings.AbstractScopedSettings.addSettingsUpdateConsumer(AbstractScopedSettings.java:437)
  at org.elasticsearch.index.IndexSettings.<init>(IndexSettings.java:816)
  at org.elasticsearch.test.IndexSettingsModule.newIndexSettings(IndexSettingsModule.java:63)
  at org.elasticsearch.test.IndexSettingsModule.newIndexSettings(IndexSettingsModule.java:48)
  at org.elasticsearch.test.IndexSettingsModule.newIndexSettings(IndexSettingsModule.java:40)
  at org.elasticsearch.index.analysis.AnalysisTestsHelper.createTestAnalysisFromSettings(AnalysisTestsHelper.java:56)
  at org.elasticsearch.index.analysis.AnalysisTestsHelper.createTestAnalysisFromSettings(AnalysisTestsHelper.java:42)
  at org.elasticsearch.index.analysis.AnalysisTestsHelper.createTestAnalysisFromClassPath(AnalysisTestsHelper.java:37)
  at org.elasticsearch.analysis.common.CJKFilterFactoryTests.setup(CJKFilterFactoryTests.java:31)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:980)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

```

**Comment by tlrx** [comment]
It fails again today: https://gradle-enterprise.elastic.co/s/ahwcvtz5bkn6w

In case it is not fixed quickly please mute the test :)

**Comment by tlrx** [comment]
@nik9000 We noticed that the periodic release tests are failing for many test with missing setting:
>     java.lang.IllegalArgumentException: Setting is not registered for key [index.time_series.end_time]

This is due to release tests passing the `-Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false"` flags which in turn prevent the registration of the `index.time_series.end_time` for these builds.

I'm leaving the resolution of this issue to you if you're OK.

**Comment by nik9000** [comment]
> I'm leaving the resolution of this issue to you if you're OK.

Sorry, yes. I'll pick it up. Was swamped yesterday.

**Comment by nik9000** [comment]
I've opened #80945 which would fix that particular issue. I've tagged it with "run the release tests" so it'll find any more release build failures. Hopefully.

**Comment by nik9000** [comment]
Should be fixed by #80945.

**Comment by tlrx** [comment]
Thanks Nik!
